helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -384,8 +384,8 @@ public synchronized void refresh(HelixDataAccessor accessor) {
     _instanceMessagesCache.updateRelayMessages(_liveInstanceCache.getPropertyMap(),
         _currentStateCache.getParticipantStatesMap());
 
-    updateIdealRuleMap();
-    updateDisabledInstances();
+    updateIdealRuleMap(getClusterConfig());
+    updateDisabledInstances(getInstanceConfigMap().values(), getClusterConfig());
```

Comment:
This seems to be non-essential too, right? The problem is in the` setClusterConfig` and `setInstanceConfig`

Reply:
This it to improve the design so we don't make such a mistake again.